src/Sets/HPolyhedron.jl

```diff
@@ -130,21 +144,79 @@ direction.
 
 ### Input
 
-- `d`      -- direction
-- `P`      -- polyhedron in constraint representation
-- `solver` -- (optional, default: `default_lp_solver(M, N)`) the backend used to
-              solve the linear program
+- `d`         -- direction
+- `P`         -- polyhedron in constraint representation
+- `algorithm` -- (optional, default: "lp") algorithm used to find the support vector
+              in the given direction; for description see the Algorithm section below.
+              Possible choices are
+
+    - `"lp"`
+    - `"halfspace_direction"`
+
+- `solver`    -- (optional, default: `default_lp_solver(M, N)`) the backend used to
+              solve the linear program for the "lp" algorithm.
 
 ### Output
 
 The support vector in the given direction.
 If a polytope is unbounded in the given direction, we throw an error.
 If a polyhedron is unbounded in the given direction, the result contains `±Inf`
 entries.
+
+### Algorithm
+
+This method applies either a LP-based approach or an approach based on direction of
+halfspaces. The default is to apply the LP-based approach. This is the safer algorithm
+as it works correctly for `HPoly` with redundant constraints, i.e. if `P` is empty or
+has redudant constraints, then the approach based on halfspace directions is not
+guaranteed to produce a correct result. 
+
+#### LP
+
+This algorithm solves the following program ``\\min_{Hx \\leq h} -d^\\mathrm{T} x``. If
+both the primal and dual are feasible, the optimal value of x is the support vector.
+If the primal is infeasible, then `P` is empty, and an error is thrown. If the dual is
+infeasible, then the primal is unbounded meaning that the polyhedron is unbounded in 
+the given direction. An infeasibility certificate is used to find the sign of the
+resuling vector of `±Inf`.
+
+#### Halfspace normal vector directions
+
+Note: This algorithm only applies if the `P` contains no redundant constraints. If not,
+this algorithm is not guaranteed to produce a correct result.
+
+This algorithm is solves the problem of finding support vectors by analyzing the normal
+vectors of halfspace constraints. The algorithm relies on the following three key insights:
+
+1. The support vector will always be a vertex or unbounded.
+2. The `dim(P)` constraints with the smallest angle to `d` will constraint the vertex ``x``
+   maximizing ``d^\\mathrm{T}x``.
+3. If the angle between `d` and the normal vector of a halfspace is greater than 
+   ``90^{\\circ}``, then that halfspace cannot constraint the optimal vertex ``x``.
+
+Following this train of thought, the algorithm finds the `dim(P)` constraints with the
+largest  ``\\cos(\\theta) = \\frac{\\langle a, d \\rangle}{\\lVert a \\rVert \\lVert d \\rVert}``
```

Comment:
I have a mismatch between "The `dim(P)` constraints with the smallest angle" above and the "largest" here. Is one of them a typo?

Reply:
No, this is not a typo. The smallest absolute angle correspond to the largest `cos(theta)`, i.e. the mapping from `abs(theta)` to `cos(theta)` for -pi <= theta <= pi is monotone, so by considering  `cos(theta)`  directly, we avoid taking the `arccos` and `abs`. Furthermore, `cos(theta) < 0` if and only if the angle is greater than 90 degrees, hence a normal vector satisfying `cos(theta) < 0` cannot bound `x` and we can filter it (to improve the speed of the topk via partial sorting).

Reply:
Okay, do you mind adding a clarifying comment to avoid future confusion? Maybe just add "(which corresponds to the smallest absolute angle)" to this sentence. Or just copy what you wrote here.

Reply:
The documentation has now been updated to explain the monotone mapping. I have also included an explanation of how the algorithm handles an unbounded support vector.